### ISSUE TITLE
Config file location on Windows

### ISSUE THREAD

**Issue by grawity** [rg3/youtube-dl#242]
Currently, youtube-dl uses Unix-specific locations (XDG_CONFIG_HOME) for finding the configuration files. On Windows, "Application Data" is a better location for this, and the proper way to find it is by using `SHGetFolderPath()`, as in [this example][1].

(Windows also provides `os.environ["APPDATA"]`, but this is somewhat less reliable, and does not exist in some uncommon situations.)

[1]: https://gist.github.com/1427281

**Comment by jaimeMF** [comment]
youtube-dl uses now `%APPDATA%\youtube-dl\config.txt` or `C:\Users\<user name>\youtube-dl.conf` [see the README](https://github.com/rg3/youtube-dl/blob/master/README.md#configuration). So I'm closing the issue.  If you think it should be reopen, please comment here. Thanks for the report!

**Comment by dogancelik** [comment]
`C:\Users\<user name>\youtube-dl.conf` is not working but `%APPDATA%` method works, can you look at this issue?

**Comment by Hrxn** [comment]
I've been using `C:\Users\<user>\youtube-dl.conf` since forever, and it never failed.

Are you absolutely positive that the filename is `youtube-dl.conf`, and not `youtube-dl.conf.txt` etc.?
Or maybe some unsupported char symbol as Username?

**Comment by dogancelik** [comment]
Yes I tested

![conf1](https://user-images.githubusercontent.com/486818/32739820-0a844892-c8b2-11e7-949f-a3344c7bb484.gif)

**Comment by photonometric** [comment]
Personally I put my "portable" programs in a custom high-level folder because the User and Appdata folders are awkward to use. I think the configuration file should be allowed in the program directory; your reference to standard locations is very well thought out, but the vast majority of "project" software like this keep their config.ini folder in the program directory. It's where I have to go to run the program, and I won't lose it if I move the program/reformat/etc and forget its config is in an arbitrary separate location. Large/complex software like MS Office or Firefox seems to be more what separate Appdata or user-folder configs is best for.

tldr it would be nice if the file would be recognized in the program's own folder, whether config.txt, youtube-dl.conf, or the more traditional config.ini :)

**Comment by canklot** [comment]
Does config file created by default ? Or is it created after a commands in youtube dl ?

**Comment by Hrxn** [comment]
You have to create that file manually..

**Comment by Naomi010Sentzke** [comment]
Still no way to have a global system wide config in Windows

**Comment by Guillaume-Garrigos** [comment]
I don't know if this may help somebody, but I tried to put the config file in 
%APPDATA%\youtube-dl\config.txt
and it did not work. Though, this location seemed strange to me since on my computer (xps 13, Win10) the configuration/profile files are stored in subfolders of AppData, usually in Roaming. So I did instead

C:\Users\MyUserName\AppData\Roaming\youtube-dl\config.txt

and it works as intended.

**Comment by Hrxn** [comment]
This should be the same path, actually.

**Comment by Guillaume-Garrigos** [comment]
Ohh I understand now. %APPDATA% is a shortcut for
C:\Users\MyUserName\AppData\Roaming\
and not
C:\Users\MyUserName\AppData\

Thanks for the hint.

**Comment by Hrxn** [comment]
Yep, exactly 😄
